### PR TITLE
Bump to latest c7n and c7n-mailer releases

### LINKED ISSUE #56: Update to use 0.9.8 Cloud Custodian release
We'd like to use a new feature added to CC 0.9.8 - can we get a bump in the Cloud Custodian dependency in Manheim with a release / docker tag? 
https://github.com/cloud-custodian/cloud-custodian/releases/tag/0.9.8.0

### PR DESCRIPTION
**Pull request by jantman**
## Description

This PR fixes #56 by bumping both c7n and c7n-mailer to their latest upstream releases. It also adds python 3.9 test runs, and bumps the base Docker image from python 3.7 to 3.9.

## Testing Done

TBD.

### CODE REVIEW

**Comment by jantman** [comment]
This PR is blocked on an upstream issue: https://github.com/cloud-custodian/cloud-custodian/issues/6359

**Comment by jantman** [comment]
A new c7n-mailer release was cut today which fixes the upstream issue, and should unblock this.

**Review by kennyg — APPROVED** [review]

**Pull request merged**
